CMakeLists.txt

```diff
@@ -112,29 +112,47 @@ include_directories(${PROJECT_BINARY_DIR})
 
 # static library
 
+option(BUILD_SHARED "Build shared library." ON)
 option(BUILD_STATIC "Build static library." OFF)
 
+if (NOT BUILD_STATIC AND NOT BUILD_SHARED)
+    message(FATAL_ERROR "BUILD_SHAREDand BUILD_STATIC cannot be both disabled")
+endif ()
+
 if (BUILD_STATIC)
     add_library(orcania_static STATIC ${LIB_SRC})
+    set_target_properties(orcania_static PROPERTIES
+            PUBLIC_HEADER "${INC_DIR}/orcania.h;${PROJECT_BINARY_DIR}/orcania-cfg.h")
     target_compile_definitions(orcania_static PUBLIC -DO_STATIC_LIBRARY)
     set_target_properties(orcania_static PROPERTIES
             OUTPUT_NAME orcania)
+    if (MSVC)
+        set_target_properties(orcania_static PROPERTIES
+                OUTPUT_NAME orcania-static)
+    endif ()
+    set(orcania_lib orcania_static)
 endif ()
 
 # shared library
 
-add_library(orcania SHARED ${LIB_SRC})
-if (NOT MSVC)
+if (BUILD_SHARED)
+    add_library(orcania SHARED ${LIB_SRC})
     set_target_properties(orcania PROPERTIES
-            COMPILE_OPTIONS -Wextra
-            PUBLIC_HEADER "${INC_DIR}/orcania.h;${PROJECT_BINARY_DIR}/orcania-cfg.h"
-            VERSION "${LIBRARY_VERSION}"
-            SOVERSION "${LIBRARY_SOVERSION}")
-endif()
-if (WIN32)
-    set_target_properties(orcania PROPERTIES SUFFIX "-${LIBRARY_VERSION_MAJOR}.dll")
+            PUBLIC_HEADER "${INC_DIR}/orcania.h;${PROJECT_BINARY_DIR}/orcania-cfg.h")
+    if (NOT MSVC)
+        set_target_properties(orcania PROPERTIES
+                COMPILE_OPTIONS -Wextra
+                VERSION "${LIBRARY_VERSION}"
```

Comment:
Question: Why is `VERSION` parameter not included in MSVC build?

Reply:
To be honest, I applied the least amount of changes to the cmake script.
`VERSION` is mostly used for versioning `.so` and `.dylib` libraries. For dll's is changing a built-in version.
See [`VERSION`](https://cmake.org/cmake/help/latest/prop_tgt/VERSION.html).

I have added a patch that improves (imho) the cmake script a bit by using targets.
I think it's still compatible with CMake 3.5.